lib/nostrum/struct/component.ex

```diff
@@ -248,14 +263,41 @@ defmodule Nostrum.Struct.Component do
   """
   @type max_values :: integer() | nil
 
+  @typedoc """
+  The minimum length of the text input. Minimum value 0, max 4000.
+
+  Valid for [Text Input](#module-text-input).
+  """
+  @type min_length :: integer() | nil
+
+  @typedoc """
+  The maximum length of the text input. Minimum value 1, max 4000.
+
+  Valid for [Text Input](#module-text-input).
+  """
+  @type max_length :: integer() | nil
+
+  @typedoc """
+  Indicates if the text input is required.
+
+  Valid for [Text Input](#module-text-input).
+  """
+  @type required :: boolean() | nil
+
+  @typedoc """
+  A pre-filled value for the text input, max 4000 characters.
+
+  Valid for [Text Input](#module-text-input).
+  """
+  @type value :: String.t() | nil
```

Comment:
Let's add `since:` metadata here:

```suggestion
  """
  @typedoc since: "0.5.1"
  @type min_length :: integer() | nil

  @typedoc """
  The maximum length of the text input. Minimum value 1, max 4000.

  Valid for [Text Input](#module-text-input).
  """
  @typedoc since: "0.5.1"
  @type max_length :: integer() | nil

  @typedoc """
  Indicates if the text input is required.

  Valid for [Text Input](#module-text-input).
  """
  @typedoc since: "0.5.1"
  @type required :: boolean() | nil

  @typedoc """
  A pre-filled value for the text input, max 4000 characters.

  Valid for [Text Input](#module-text-input).
  """
  @typedoc since: "0.5.1"
  @type value :: String.t() | nil
```